File: ingwang/programmers-2/카펫.java
class Solution33 {
    public static  int[] solution(int brown, int yellow) {
        int[] answer = new int[2];

        for (int i = 3; i <= (brown+yellow); i++) {
            int y=i;
            int x = (brown+yellow)/y;

            if(x<3){
                continue;
            }
            if (x>=y){
                if((x-2)*(y-2)==yellow){
                    answer[0] = x;
                    answer[1]= y;
                    break;
                }
            }
        }


         return answer;
   }
 
         
     public static void main(String[] args) {
         System.out.println(solution(4,5));
     }
 }   

File: soyeong/programmers-2/카펫.py
def solution(brown, yellow):
    # 가로 a, 세로 b
    # Brown = 2a + 2b - 4
    # Yellow = (a-2)(b-2) = ab - (2a + 2b - 4)
    # Yellow + Brown = ab

    answer = []
    num_ab = brown + yellow
    for a in range(1, num_ab + 1):
        if num_ab % a == 0:
            b = num_ab / a
            if (a >= b) and ((2 * a + 2 * b - 4) == brown):
                return [a, b]
    return answer